src/main/scala/com/fasterxml/jackson/module/scala/introspect/ScalaAnnotationIntrospectorModule.scala

```diff
@@ -114,9 +117,54 @@ object ScalaAnnotationIntrospector extends NopAnnotationIntrospector
       JsonCreator.Mode.PROPERTIES
     } else null
   }
+
+  class ScalaValueInstantiator(delegate: StdValueInstantiator, config: DeserializationConfig, descriptor: BeanDescriptor) extends StdValueInstantiator(delegate) {
+
+    private val overriddenConstructorArguments = {
+      val args = delegate.getFromObjectArguments(config)
+
+      args.map {
+        case creator: CreatorProperty =>
+          // Locate the constructor param that matches it
+          descriptor.properties.find(_.param.exists(_.index == creator.getCreatorIndex)) match {
+            case Some(PropertyDescriptor(name, Some(ConstructorParameter(_, _, Some(defaultValue))), _, _, _, _, _)) =>
+              creator.withNullProvider(new NullValueProvider {
+                override def getNullValue(ctxt: DeserializationContext): AnyRef = defaultValue()
+
+                override def getNullAccessPattern: AccessPattern = AccessPattern.DYNAMIC
+              })
+            case _ => creator
+          }
+        case other => other
+      }
+    }
+
+    override def getFromObjectArguments(config: DeserializationConfig): Array[SettableBeanProperty] = {
+      overriddenConstructorArguments
+    }
+  }
+
+  override def findValueInstantiator(config: DeserializationConfig, beanDesc: BeanDescription,
+    defaultInstantiator: ValueInstantiator): ValueInstantiator = {
+
+    if (isMaybeScalaBeanType(beanDesc.getBeanClass)) {
+
+      val descriptor = _descriptorFor(beanDesc.getBeanClass)
+      if (descriptor.properties.exists(_.param.exists(_.defaultValue.isDefined))) {
+        defaultInstantiator match {
+          case std: StdValueInstantiator =>
+            new ScalaValueInstantiator(std, config, descriptor)
+          case other =>
+            throw new IllegalArgumentException("Cannot customise a non StdValueInstantiatiator: " + other.getClass)
```

Comment:
The Kotlin module does exactly the same thing here, and it's been doing that for 4 years presumably with no problems, so I think we're fine to fail too:

https://github.com/FasterXML/jackson-module-kotlin/blob/3eb66acf2638dfd99a53a3d6e1389844b5caed2e/src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinValueInstantiator.kt#L174